features/logout/impl/src/main/kotlin/io/element/android/features/logout/impl/LogoutPresenter.kt

```diff
@@ -47,8 +52,15 @@ class LogoutPresenter @Inject constructor(
             mutableStateOf(Async.Uninitialized)
         }
 
-        val backupUploadState: BackupUploadState by remember {
-            encryptionService.waitForBackupUploadSteadyState()
+        val secureStorageFlag by featureFlagService.isFeatureEnabledFlow(FeatureFlags.SecureStorage)
+            .collectAsState(initial = runBlocking { featureFlagService.isFeatureEnabled(FeatureFlags.SecureStorage) })
+
+        val backupUploadState: BackupUploadState by remember(secureStorageFlag) {
+            if (secureStorageFlag) {
+                encryptionService.waitForBackupUploadSteadyState()
+            } else {
+                flowOf(BackupUploadState.Done)
```

Comment:
Actually, I'd let the `EncryptionService` handle this, so we don't have to check that in Presenters.

Reply:
I thought about doing like this, but I prefer to avoid this, since we had to give the `FeatureFlagService` to the SDK which is a bit strange. We may iterate on this later.

Reply:
Our `matrix` module is not really a sdk. I see what you mean, but it's not really the responsability of the `Presenter` neither :/